diff --git a/third-party/wslay/includes/wslay/wslay.h b/third-party/wslay/includes/wslay/wslay.h
index 2fde81a4..22c03818 100644
--- a/third-party/wslay/includes/wslay/wslay.h
+++ b/third-party/wslay/includes/wslay/wslay.h
@@ -33,6 +33,10 @@ extern "C" {
 #include <stdlib.h>
 #include <sys/types.h>
 
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
 
 /*
  * wslay/wslayver.h is generated from wslay/wslayver.h.in by
