diff --git a/thirdparty/bullet/LinearMath/btVector3.h b/thirdparty/bullet/LinearMath/btVector3.h
index 9da16277..e15a8ab3 100644
--- a/thirdparty/bullet/LinearMath/btVector3.h
+++ b/thirdparty/bullet/LinearMath/btVector3.h
@@ -1092,6 +1092,11 @@ public:
 		mVec128 = rhs.mVec128;
 	}
 
+	SIMD_FORCE_INLINE btVector4(const btVector4& rhs)
+	{
+		mVec128 = rhs.mVec128;
+	}
+
 	SIMD_FORCE_INLINE btVector4&
 	operator=(const btVector4& v)
 	{
